diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -301,6 +303,19 @@ fun ASTNode?.isAccessibleOutside(): Boolean =
             true
         }
 
+fun ASTNode.hasSuppress(warningName: String): Boolean {
+    return if (elementType == FILE) {
+        findAllNodesWithSpecificType(ANNOTATION_ENTRY).any {
+            it.text.contains("Suppress(\"$warningName\")")
+        }
+    } else {
+        parent({
+            it.findAllNodesWithSpecificType(ANNOTATION_ENTRY).any {
```

Comment:
This way suppress, e.g., on one class will also suppress warning on another. I think you should look for parent's modifier list and search annotation there.